lib/mix/test/mix/tasks/compile.elixir_test.exs

```diff
@@ -438,4 +438,50 @@ defmodule Mix.Tasks.Compile.ElixirTest do
       end)
     end
   end
+
+  test "does not use incorrect line number when error originates in another file" do
+    in_fixture "no_mixfile", fn ->
+      File.write!("lib/a.ex", """
+      defmodule A do
+        def fun(arg), do: arg / 2
+      end
+      """)
+
+      File.write!("lib/b.ex", """
+      defmodule B do
+        def fun(arg) do
+          A.fun(arg)
+          :ok
+        end
+      end
+      B.fun(:not_a_number)
+      """)
+
+      import ExUnit.CaptureIO
```

Comment:
When you move this test to paralle_compiler_test.exs, you will no longer need this import.